### PR TITLE
Dynamic octokit instantiation based on auth status

### PR DESCRIPTION
**Pull request by nickcastel50**
* Dynamic octokit instantiation based on auth status, we probably don't want to use the proxy for unearthed requests due to rate limiting.

### CODE REVIEW

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]
Let's figure out a way to test this.  Should need some new hooks in msw?  for new non-auth'd paths

File: src/net/github/OctokitExport.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+export function initializeOctoKit(authed) {
+    if (authed) {
+        octokit = new Octokit({
+        baseUrl: GITHUB_BASE_URL_AUTHED,
+        userAgent: `bldrs/${PkgJson.version}`,
+      })
+    } else {
+        octokit = new Octokit({
+            baseUrl: 'https://api.github.com',
```
Comment (pablo-mayrgundter) [inline src/net/github/OctokitExport.js:25]: Pretty sure this should be from process.env

Comment (nickcastel50) [reply src/net/github/OctokitExport.js]: updated

File: src/net/github/OctokitExport.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Initialize an instance of Octokit depending on auth status.
+ */
+export function initializeOctoKit(authed) {
+    octokit = new Octokit({
+    baseUrl: authed ? GITHUB_BASE_URL_AUTHED : GITHUB_BASE_URL_UNAUTHED,
+    userAgent: `bldrs/${PkgJson.version}`,
+    })
+}
```
Comment (pablo-mayrgundter) [inline src/net/github/OctokitExport.js:22]: Please turn this into two functions, initializeOctoKitAuthed and initializeOctoKitUnauthed

**Review by pablo-mayrgundter — APPROVED** [review]

File: src/net/github/OctokitExport.js (hunk 2 above)
Comment (nickcastel50) [reply src/net/github/OctokitExport.js:32]: PTAL

**Pull request merged**